### ISSUE TITLE
MALOC modernization

### ISSUE THREAD

**Issue by sobolevnrm** [Electrostatics/apbs#23]
It's no longer needed and takes extra effort to maintain.

**Comment by sobolevnrm** [comment]
The most likely modernization route here is just to delete the library.

**Comment by ashermancinelli** [comment]
This issue is being developed in branch `remove-maloc-dev`

**Comment by ashermancinelli** [comment]
@sobolevnrm In the branch mentioned above, I've removed maloc as a dependency. I haven't changed too much, just copied five header/source pairs into another subdirectory of `src`. Would you like to iteratively replace the remnants of maloc as well or are we content with this solution?

CC @intendo

**Comment by sobolevnrm** [comment]
I don't think we need to rewrite MALOC.  If native functionality is available to replace MALOC functionality, then that's all we need.

**Comment by ashermancinelli** [comment]
So would you like to leave a few components of maloc in the repo like in `remove-maloc-dev` branch? If we go the native functionality route, we'd probably want to refactor pieces of apbs source into C++. Biggest MALOC component used is Vio/socket stuff I think, we could replace all that with std::string. APBS/MALOC sources seem compile-able in C++ already anyways.

**Comment by sobolevnrm** [comment]
I think it's best to replace whatever we can with native functionality and
then see what's left?
